src/plugins_types/integer.c

```diff
@@ -28,54 +29,105 @@
 #include "compat.h"
 #include "plugins_internal.h" /* LY_TYPE_*_STR */
 
+/**
+ * @page howtoDataLYB LYB Binary Format
+ * @subsection howtoDataLYBTypesInteger (u)int(8/16/32/64) (built-in)
+ *
+ * | Size (B) | Mandatory | Type | Meaning |
+ * | :------  | :-------: | :--: | :-----: |
+ * | 1/2/4/8 | yes | pointer to the specific integer type | integer value |
+ */
+
 API LY_ERR
 lyplg_type_store_int(const struct ly_ctx *ctx, const struct lysc_type *type, const void *value, size_t value_len,
-        uint32_t options, LY_VALUE_FORMAT UNUSED(format), void *UNUSED(prefix_data), uint32_t hints,
+        uint32_t options, LY_VALUE_FORMAT format, void *UNUSED(prefix_data), uint32_t hints,
         const struct lysc_node *UNUSED(ctx_node), struct lyd_value *storage, struct lys_glob_unres *UNUSED(unres),
         struct ly_err_item **err)
 {
     LY_ERR ret = LY_SUCCESS;
     int64_t num;
+    size_t lyb_size;
     int base = 1;
-    char *str;
+    char *canon;
     struct lysc_type_num *type_num = (struct lysc_type_num *)type;
 
-    /* check hints */
-    ret = lyplg_type_check_hints(hints, value, value_len, type->basetype, &base, err);
-    LY_CHECK_GOTO(ret != LY_SUCCESS, cleanup);
+    /* clear storage */
+    memset(storage, 0, sizeof *storage);
 
-    switch (type->basetype) {
-    case LY_TYPE_INT8:
-        ret = lyplg_type_parse_int("int8", base, INT64_C(-128), INT64_C(127), value, value_len, &num, err);
-        break;
-    case LY_TYPE_INT16:
-        ret = lyplg_type_parse_int("int16", base, INT64_C(-32768), INT64_C(32767), value, value_len, &num, err);
-        break;
-    case LY_TYPE_INT32:
-        ret = lyplg_type_parse_int("int32", base, INT64_C(-2147483648), INT64_C(2147483647), value, value_len, &num, err);
-        break;
-    case LY_TYPE_INT64:
-        ret = lyplg_type_parse_int("int64", base, INT64_C(-9223372036854775807) - INT64_C(1),
-                INT64_C(9223372036854775807), value, value_len, &num, err);
-        break;
-    default:
-        LOGINT(ctx);
-        ret = LY_EINT;
-    }
+    if (format == LY_VALUE_LYB) {
```

Comment:
What exactly is the use? It would work only for built-in types and only those with fixed sizes, so an array seems unsafe. Also, it could return misleading values for derived types with specific plugins using values of a different size.

Reply:
to get rid of those long switches, it can be just internal/local. Just an idea to consider.

Reply:
Okay, an internal array can be used.